internal/resources/metal/vlan/resource_test.go

```diff
@@ -0,0 +1,243 @@
+		}
+
+		client := acceptance.TestAccProvider.Meta().(*config.Config).Metal
+
+		foundVlan, _, err := client.ProjectVirtualNetworks.Get(rs.Primary.ID, nil)
+		if err != nil {
+			return err
+		}
+		if foundVlan.ID != rs.Primary.ID {
+			return fmt.Errorf("Record not found: %v - %v", rs.Primary.ID, foundVlan)
+		}
+
+		*vlan = *foundVlan
+
+		return nil
+	}
+}
+
+func testAccMetalVlanCheckDestroyed(s *terraform.State) error {
+	client := acceptance.TestAccProvider.Meta().(*config.Config).Metal
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "equinix_metal_vlan" {
+			continue
+		}
+		if _, _, err := client.ProjectVirtualNetworks.Get(rs.Primary.ID, nil); err == nil {
+			return fmt.Errorf("Metal Vlan still exists")
+		}
+	}
+
+	return nil
+}
+
+func TestAccMetalVlan_importBasic(t *testing.T) {
+	rs := acctest.RandString(10)
+	metro := "sv"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acceptance.TestAccPreCheckMetal(t) },
+		ExternalProviders:        acceptance.TestExternalProviders,
+		ProtoV5ProviderFactories: acceptance.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccMetalVlanCheckDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckMetalVlanConfig_metro(rs, metro, "tfacc-vlan"),
+			},
+			{
+				ResourceName:      "equinix_metal_vlan.foovlan",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccMetalVlan_facility_to_metro(t *testing.T) {
+	var vlan packngo.VirtualNetwork
+	rs := acctest.RandString(10)
+	metro := "sv"
+	facility := "sv15"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acceptance.TestAccPreCheckMetal(t) },
+		ExternalProviders:        acceptance.TestExternalProviders,
+		ProtoV5ProviderFactories: acceptance.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccMetalVlanCheckDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckMetalVlanConfig_facility(rs, facility, "tfacc-vlan"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMetalVlanExists("equinix_metal_vlan.foovlan", &vlan),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "description", "tfacc-vlan"),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "metro", metro),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "facility", facility),
+				),
+			},
+			{
+				Config:             testAccCheckMetalVlanConfig_metro(rs, metro, "tfacc-vlan"),
+				ExpectNonEmptyPlan: false,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMetalVlanExists("equinix_metal_vlan.foovlan", &vlan),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "description", "tfacc-vlan"),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "metro", metro),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "facility", facility),
```

Comment:
Actually I think your original test expectation was correct; the API will still return `facility` because the VLAN was created in a facility, so the attribute should stay the same after migrating.